aquadoggo/src/graphql/replication/mod.rs

```diff
@@ -0,0 +1,267 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use anyhow::Error as AnyhowError;
+use async_graphql::connection::{query, Connection, CursorType, Edge, EmptyFields};
+use async_graphql::Object;
+use async_graphql::*;
+use mockall_double::double;
+use p2panda_rs::entry::decode_entry;
+use p2panda_rs::storage_provider::traits::EntryStore;
+use tokio::sync::Mutex;
+
+use crate::db::stores::StorageEntry;
+
+pub mod aliased_author;
+pub mod author;
+pub mod client;
+pub mod context;
+pub mod entry;
+pub mod entry_and_payload;
+pub mod entry_hash;
+pub mod log_id;
+pub mod payload;
+pub mod public_key;
+pub mod sequence_number;
+pub mod single_entry_and_payload;
+
+#[cfg(test)]
+mod testing;
+
+pub use aliased_author::AliasedAuthor;
+pub use author::{Author, AuthorOrAlias};
+
+#[double]
+pub use context::Context as ReplicationContext;
+pub use entry::Entry;
+pub use entry_and_payload::EntryAndPayload;
+pub use entry_hash::EntryHash;
+pub use log_id::LogId;
+pub use payload::Payload;
+pub use public_key::PublicKey;
+pub use sequence_number::SequenceNumber;
+pub use single_entry_and_payload::SingleEntryAndPayload;
+
+#[derive(Debug)]
+/// The root graphql object for replication
+pub struct ReplicationRoot<ES> {
+    entry_store: PhantomData<ES>,
```

Comment:
Yeah exactly. 

If you don't use the `PhantomData` type the compiler will complain because you don't actually use the type in the struct.

I did it this way so that `ReplicationRoot` methods didn't have to be generic over `EntryStore`